Tribler/Core/APIImplementation/LaunchManyCore.py

```diff
@@ -594,31 +595,33 @@ def early_shutdown(self):
         """ Called as soon as Session shutdown is initiated. Used to start
         shutdown tasks that takes some time and that can run in parallel
         to checkpointing, etc.
+        :returns a Deferred that will fire once all dependencies acknowledge they have shutdown.
         """
         self._logger.info("tlm: early_shutdown")
 
         self.cancel_all_pending_tasks()
 
         # Note: sesslock not held
+        wait_defer_list = []
         self.shutdownstarttime = timemod.time()
         if self.torrent_checker:
-            self.torrent_checker.shutdown()
+            wait_defer_list.append(self.torrent_checker.shutdown())
```

Comment:
Since shutting down the torrentchecker is now done asynchronously with Deferrds the `early_shutdown` should also be able to handle this. As we want to streamline the shut down procedure as per #1848 I figured I make it so that once other components start returning Deferrds, the early_shutdown can handle it already.

But if you prefer it the current way, then I'll revert those maybeDeferreds except for the torrentchecker shutdown of course.